manifests/agent_status_enable.pp

```diff
@@ -0,0 +1,29 @@
+# @summary Enables the execution of agent_status_check fact
+#
+# Adding this class will enable the execution of the agent_status_check fact, 
+# This allows the fact to be targeted to a specific agent or group of agents
+#
+# @example
+#   include pe_status_check::agent_status_enable
+# @param [Boolean] agent_status_enabled
+#  Flag to enable or disable agent_status_check fact
+class pe_status_check::agent_status_enable (
+  Boolean       $agent_status_enabled               = true,
+){
+  $agent_status_enabled_file = $facts['os']['family'] ? {
+    'windows' => "${facts['common_appdata']}/PuppetLabs/puppet/cache/state/status_check_enable",
+    default   => '/opt/puppetlabs/puppet/cache/state/status_check_enable',
+  }
+if $agent_status_enabled {
```

Comment:
I am not sure why, but it feels like this `if/else` should be a ternary like the following.
```
$agent_status_enabled_file_ensure = $agent_status_enabled ? {
  true     => 'file',
  default => 'absent',
}
file { $agent_status_enabled_file:
  ensure => $agent_status_enabled_file_ensure,
  mode   => '0664',
}
```

Reply:
felt like an odd choice but its a copy paste, i can change it